include/zephyr/devicetree.h

```diff
@@ -3697,6 +3741,19 @@
  */
 #define DT_INST_REG_ADDR(inst) DT_INST_REG_ADDR_BY_IDX(inst, 0)
 
+/**
+ * @brief 64-bit version of DT_INST_REG_ADDR()
+ *
+ * This macro version adds the appropriate suffix for 64-bit unsigned
+ * integer literals.
+ * Note that this macro is equivalent to DT_INST_REG_ADDR() in
+ * linker context.
```

Comment:
```suggestion
 * linker/ASM context.
```
? (see my other comments)